CARDS-1507: As a patient, I have access to a high-level interpretation of my survey responses

acrowthe opened the pull request:
- Implement review questions for AUDIT-C, PHQ9 and GAD-7 questionnaires
  - Hidden when filling out forms, visible on the final patient UI page
  
Note: the implementation assumes that the `sex` variable in AUDIT-C is populated at the time of creating the form, from the variable with the same name from Patient information. This will be done as part of #839 . For testing purposes, set that variable manually to one of the supported values `Male`, `Female`, `Other`.

acrowthe committed:
CARDS-1507: As a patient, I have access to a high-level interpretation of my survey responses
- Implement review questions for AUDIT-C, PHQ9 and GAD-7 questionnaires
  - Hidden when filling out forms, visible on the final patient UI page

acrowthe committed:
CARDS-1507: As a patient, I have access to a high-level interpretation of my survey responses
- Improve hiding/showing of review mode questions and sections

acrowthe commented:
Improved how `<Form>` handles review mode questions and sections by adding a `displayMode="review"` to Sections and ComputedQuestions

sashaandjic commented:
Should we hide the list of pre-op surveys once they are complete?
![image](https://user-images.githubusercontent.com/23243244/149415962-2211db16-42d7-4101-bdaf-9b4f63892d6e.png)

marta- commented:
> Should we hide the list of pre-op surveys once they are complete?

Agreed.

sashaandjic commented:
In the JIRA task , the highlighted text is bolded.
![image](https://user-images.githubusercontent.com/23243244/149417814-b52425e1-cf5e-430b-9f91-0f72c2d83a6b.png)

marta- commented on proms-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/PHQ9.xml:
```diff
@@ -273,6 +273,90 @@ Objectively determines severity of initial symptoms, and also monitors symptom c
+			</property>
+			<property>
+				<name>text</name>
+				<value>PHQ-9 Depression Questionnaire</value>
```
```suggestion
				<value>PHQ-9 Interpretation</value>
```

marta- commented on proms-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/AUDITC.xml:
```diff
@@ -127,6 +127,72 @@ disorder. AUDIT-C has been validated in mental health and primary care clinics.
+			</property>
+			<property>
+				<name>text</name>
+				<value>AUDIT-C Alcohol Use Questionnaire</value>
```
```suggestion
				<value>AUDIT-C Interpretation</value>
```

marta- commented on proms-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/GAD7.xml:
```diff
@@ -248,6 +248,47 @@ Objectively determine initial symptoms severity and monitor symptom changes/effe
+			</property>
+			<property>
+				<name>text</name>
+				<value>GAD-7 Anxiety Questionnaire</value>
```
```suggestion
				<value>GAD-7 Interpretation</value>
```

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/Form.jsx:
```diff
@@ -124,6 +124,7 @@ function Form (props) {
   const formURL = `/Forms/${id}`;
   const urlBase = "/content.html";
   const isEdit = window.location.pathname.endsWith(".edit") || mode == "edit";
+  const isReview = window.location.pathname.endsWith(".review") || mode == "review";
```
Let's replace `review` with `summary` everywhere.

sashaandjic commented:
So we will not be looking if patient is F/M/O?
![image](https://user-images.githubusercontent.com/23243244/149420823-54e2568a-ed29-4d29-a4a4-2ce1d19854a8.png)

marta- commented:
> So we will not be looking if patient is F/M/O?

This PR assumes there's a hidden question in AUDITC where the sex from Patient information is copied over. @acrowthe  Are the changes that populate this variable in 1474?

sashaandjic commented:
The button run away...
![image](https://user-images.githubusercontent.com/23243244/149421493-e1f0ea1d-feb9-4687-acf7-9524622b4e0f.png)

marta- commented on proms-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/AUDITC.xml:
```diff
@@ -127,6 +127,72 @@ disorder. AUDIT-C has been validated in mental health and primary care clinics.
+						(
+							"male" === @{sex:-""}.toLowerCase() ? maleText
+								: "female" === @{sex:-""}.toLowerCase() ? femaleText
+									: "For Males:\n\n" + maleText + "\n\n For Females:\n\n" + femaleText
```
```suggestion
									: "If you are a man:\n\n" + maleText + "\n\n If you are a woman:\n\n" + femaleText
```

acrowthe commented:
> @acrowthe Are the changes that populate this variable in 1474?
They have not been implemented anywhere yet. The logical place to put them IMO is in the visit change listener implemented by 1474, but I don't want to make any changes there until Sergiu's rebase gets pushed

sashaandjic commented:
QUESTION: Should patients be seeing the scores while doing surveys?
![image](https://user-images.githubusercontent.com/23243244/149421624-489ea8a8-bf0e-4908-8ef5-703b1ceef167.png)

acrowthe commented:
> QUESTION: Should patients be seeing the scores while doing surveys?

If you signed in as admin, you should see those scores. But if you went through the process to generate a patient token and signed in with that and are still seeing the scores, that is a problem.

marta- commented:
> QUESTION: Should patients be seeing the scores while doing surveys?

No, and it will be fixed by implementing real patient authentication. Right now the user that is filling out the surveys (either admin or test user) still has access to these clinician-facing interpretations.

marta- commented on proms-resources/frontend/src/main/frontend/src/QuestionnaireSet.jsx:
```diff
@@ -346,13 +346,42 @@ function QuestionnaireSet(props) {
-      <Typography variant="h4">Thank you</Typography>
-      :
-      <Fab variant="extended" color="primary" onClick={() => {setCrtStep(-1)}}>Update answers</Fab>
+      <>
+        <Typography color="error">Your answers are incomplete. Please return to the main screen and check for any mandatory questions you may have missed.</Typography>
+        <Fab variant="extended" color="primary" onClick={() => {setCrtStep(-1)}}>Update answers</Fab>
+      </>
```
> The button run away...

This change is causing [the awkward button positioning](https://github.com/data-team-uhn/cards/pull/850#issuecomment-1012589019).

sashaandjic commented:
Very minor thing: When there is only 1 survey, the number is not displayed and as a result the alignment is a bit off.
![image](https://user-images.githubusercontent.com/23243244/149423774-be63054f-924f-482d-917d-c517d00180ab.png)
![image](https://user-images.githubusercontent.com/23243244/149423803-0719d7a5-fec8-4a5a-b579-07cff558147b.png)

marta- commented:
> Very minor thing: When there is only 1 survey, the number is not displayed and as a result the alignment is a bit off. 

@sashaandjic please make a separate task, I'll look into it

acrowthe committed:
CARDS-1507: As a patient, I have access to a high-level interpretation of my survey responses
Address comments:
- Rename "review" to "summary"
- Remove survey list when completed
- Adjust PHQ-9 text and formatting
- Improve interpretation wording

acrowthe commented:
I've addressed all the comments but:
- the FAB being misaligned (and now also the surveys on the same page)
- The issue with there being just 1 survey

sashaandjic commented:
build fails
![image](https://user-images.githubusercontent.com/23243244/149433859-435e588c-283a-44f3-bd45-dd5415d87dbe.png)

marta- commented:
> build fails 

"Too many filed with unapproved license" -> the test CSVs. We should exclude CSVs from license check. Meanwhile, you can either build with `-Pquick` or move the CSVs outside of the sources folder.

sashaandjic commented:
Please remove the list of surveys once their results are available.
![image](https://user-images.githubusercontent.com/23243244/149438204-48e52d5d-f206-4974-885e-b0201195262b.png)

sashaandjic commented:
The highlighted text should be bolded.
![image](https://user-images.githubusercontent.com/23243244/149438405-14594f96-abee-48d5-b896-38273a0a3945.png)

marta- commented:
@sashaandjic The screenshots above look like they're from an older version. The changes have in fact been made.

sashaandjic reviewed (approved):
tested - all good
![image](https://user-images.githubusercontent.com/23243244/149556059-68c889fa-5342-4618-9705-4b82d54a3361.png)

marta- reviewed (approved):
Approving provided that some fixes for 1507 are already in the 1506 branch.

sdumitriu reviewed (approved):
The summaries follow paper logic. We already know the score, why do we need to say "if your score is X, then Y", and not "your score is X, so Y", or just "Y". Then we wouldn't have to show the score, which is also inconsistent since we don't show the score for EQ-5D.

"If your symptoms result in a score of ..., make sure to bring them up to your provider at your next appointment". I think the doctor should see the score and act on it, whether the patient does or does not broach the subject.

We don't have summaries for SC and EQ-5D, should we add something?
